src/pubsub.c

```diff
@@ -387,6 +389,7 @@ void pubsubShardUnsubscribeAllChannelsInSlot(unsigned int slot) {
             if (clientTotalPubSubSubscriptionCount(c) == 0) {
                 unmarkClientAsPubSub(c);
             }
+            listDelNode(clients, ln);
```

Comment:
@CharlesChen888  please also take a look

Reply:
Yes, we do need a listDelNode here. I was using a listIterator at the beginning, somehow the listNext became listFirst during refactoring.